mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -471,15 +471,16 @@ public MessageUid apply(SearchResult input) {
     @Override
     public List<MessageId> search(MailboxSession session, MultimailboxesSearchQuery searchQuery, long limit) throws MailboxException {
         Preconditions.checkArgument(session != null, "'session' is mandatory");
-        return FluentIterable.from(searchMultimap(searchQuery))
-            .transform(new Function<SearchResult, MessageId>() {
-                @Override
-                public MessageId apply(SearchResult input) {
-                    return input.getMessageId().get();
-                }
-            })
-            .limit(Long.valueOf(limit).intValue())
-            .toList();
+        return ImmutableList.copyOf(

```

Comment:
Shouldn't you transform toSet and then apply the limit because it seems to me that you can return something inferior to limit even if there were limit or more element.
Is it ok too that you are loosing the order ?